### PR TITLE
FIX: Error plot psd fmax

### LINKED ISSUE #6862: BUG, VIZ: raw.plot_psd fmax param ignored when plotting
the `fmax` parameter should be used to set the right-hand xlim of the plot, but it is not. In the plots below, the right-hand xlim is determined by the `sfreq` even though `fmax=350` is specified for both plots.

```
import os
import mne
sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
                                    'sample_audvis_raw.fif')
raw = mne.io.read_raw_fif(sample_data_raw_file)
raw.crop(0, 60).load_data() 

raw_downsampled = raw.copy().resample(sfreq=100)

for data in (raw, raw_downsampled):
    fig = data.plot_psd(fmax=350)
```
Figure_1

Figure_2

### PR DESCRIPTION
**Pull request by drammock**
Closes #6862

### CODE REVIEW

File: mne/viz/utils.py [hunk 1]
```diff
@@ -3060,6 +3060,14 @@ def _convert_psds(psds, dB, estimate, scaling, unit, ch_names):
     return ylabel
 
 
+def _check_psd_fmax(inst, fmax):
+    """Make sure requested fmax does not exceed Nyquist frequency."""
+    if np.isfinite(fmax) and (fmax > inst.info['sfreq'] / 2):
+        raise ValueError('Requested fmax ({} Hz) must be not exceed 2× the '
+                         'sampling frequency of the data ({}).'
+                         .format(fmax, 2 * inst.info['sfreq']))
```
Comment (agramfort) [inline mne/viz/utils.py]: it should be 0.5 sfreq. 100Hz sampling has a max freq of 50Hz

Comment (drammock) [reply mne/viz/utils.py]: @agramfort thanks for catching that. Fixed (also rebased and force-pushed to get travis working); ready for review.

File: mne/viz/tests/test_raw.py [hunk 2]
```diff
@@ -411,6 +411,10 @@ def test_plot_raw_psd():
     raw.plot_psd(reject_by_annotation=True)
     raw.plot_psd(reject_by_annotation=False)
 
+    # test fmax value checking
+    with pytest.raises(ValueError, match='must be not exceed 2× the sampling'):
```
Comment (agramfort) [inline mne/viz/tests/test_raw.py]:
```suggestion
    with pytest.raises(ValueError, match='must be not exceed one half the sampling'):
```

File: mne/viz/tests/test_raw.py [hunk 3]
```diff
@@ -411,6 +411,10 @@ def test_plot_raw_psd():
     raw.plot_psd(reject_by_annotation=True)
     raw.plot_psd(reject_by_annotation=False)
 
+    # test fmax value checking
+    with pytest.raises(ValueError, match='must be not exceed one half the sampling'):
```
Comment (drammock) [inline mne/viz/tests/test_raw.py]:
```suggestion
    with pytest.raises(ValueError, match='must not exceed one half the sampling'):
```

File: mne/viz/utils.py [hunk 4]
```diff
@@ -3077,6 +3077,14 @@ def _convert_psds(psds, dB, estimate, scaling, unit, ch_names):
     return ylabel
 
 
+def _check_psd_fmax(inst, fmax):
+    """Make sure requested fmax does not exceed Nyquist frequency."""
+    if np.isfinite(fmax) and (fmax > inst.info['sfreq'] / 2):
+        raise ValueError('Requested fmax ({} Hz) must be not exceed one half '
```
Comment (drammock) [inline mne/viz/utils.py]:
```suggestion
        raise ValueError('Requested fmax ({} Hz) must not exceed one half '
```

**Comment by larsoner** [comment]
Thanks @drammock

**Pull request merged**